p2p/pex_reactor.go

```diff
@@ -293,6 +309,160 @@ func (r *PEXReactor) ensurePeers() {
 	}
 }
 
+// Explores the network searching for more peers. (continuous)
+// Seed/Crawler Mode causes this node to quickly disconnect
+// from peers, except other seed nodes.
+func (r *PEXReactor) seedCrawlerMode() {
+	// Do an initial crawl
+	r.crawlPeers()
+
+	// Fire periodically
+	ticker := time.NewTicker(5 * time.Second)
+
+	for {
+		select {
+		case <-ticker.C:
+			r.attemptDisconnects()
+			r.crawlPeers()
+		case <-r.Quit:
+			return
+		}
+	}
+}
+
+// crawlStatus handles temporary data needed for the
+// network crawling performed during seed/crawler mode.
+type crawlStatus struct {
+	// The remote address of a potential peer we learned about
+	Addr *NetAddress
+
+	// Not empty if we are connected to the address
+	PeerID string
+
+	// The last time we attempt to reach this address
+	LastAttempt time.Time
+
+	// The last time we successfully reached this address
+	LastSuccess time.Time
+}
+
+// oldestAttempt implements sort.Interface for []crawlStatus
+// based on the LastAttempt field.
+type oldestAttempt []crawlStatus
+
+func (oa oldestAttempt) Len() int           { return len(oa) }
+func (oa oldestAttempt) Swap(i, j int)      { oa[i], oa[j] = oa[j], oa[i] }
+func (oa oldestAttempt) Less(i, j int) bool { return oa[i].LastAttempt.Before(oa[j].LastAttempt) }
+
+// getCrawlStatus returns addresses of potential peers that we wish to validate.
+// NOTE: The status information is ordered as described above.
+func (r *PEXReactor) getCrawlStatus() []crawlStatus {
+	var oa oldestAttempt
+
+	addrs := r.book.List()
```

Comment:
is there a reason why we want all addresses and not `GetSelection`?

Reply:
GetSelection only returns some of the known addresses, instead of allowing the crawler to reach out to all the known addresses and ensure we're increasing its visibility of the network.